src/aiidalab_qe/plugins/xas/setting.py

```diff
@@ -0,0 +1,244 @@
+        self.core_hole_pseudos = pseudo_data_dict["pbe"]["core_hole_pseudos"]["1s"]
+        self.core_wfc_data = pseudo_data_dict["pbe"]["core_wavefunction_data"]
+
+        self.element_and_ch_treatment = ipw.VBox(layout=ipw.Layout(width="100%"))
+
+        self.structure_type = ipw.ToggleButtons(
+            options=[
+                ("Molecule", "molecule"),
+                ("Crystal", "crystal"),
+            ],
+            value="crystal",
+        )
+        self.supercell_min_parameter = ipw.FloatText(
+            value=8.0,
+            description="The minimum cell length (Å):",
+            disabled=False,
+            style={"description_width": "initial"},
+        )
+
+        self.children = [
+            # self.structure_title,
+            # self.structure_help,
+            # ipw.HBox(
+            #     [self.structure_type],
+            # ),
+            self.element_selection_title,
+            self.element_selection_help,
+            ipw.HBox([self.element_and_ch_treatment], layout=ipw.Layout(width="95%")),
+            self.supercell_title,
+            self.supercell_help,
+            ipw.HBox(
+                [self.supercell_min_parameter],
+            ),
+        ]
+
+        super().__init__(**kwargs)
+
+    def get_panel_value(self):
+        """Return a dictionary with the input parameters for the plugin."""
+
+        elements_list = []
+        core_hole_treatments = {}
+        for entry in self.element_and_ch_treatment.children:
+            if entry.children[0].value is True:
+                element = entry.children[0].description
+                ch_treatment = entry.children[1].value
+                elements_list.append(element)
+                core_hole_treatments[element] = ch_treatment
+
+        parameters = {
+            "core_hole_treatments": core_hole_treatments,
+            "elements_list": elements_list,
+            "structure_type": self.structure_type.value,
+            "gipaw_pseudo": self.gipaw_pseudos,
+            "ch_pseudo": self.core_hole_pseudos,
+            "core_wfc_data": self.core_wfc_data,
+        }
+        return parameters
+
+    def set_panel_value(self, input_dict):
+        """Load a dictionary with the input parameters for the plugin."""
+
+        ch_pseudos = self.core_hole_pseudos
+        structure = self.input_structure
+        available_elements = [k for k in ch_pseudos]
+        elements_to_select = sorted(
+            [
+                kind.symbol
+                for kind in structure.kinds
+                if kind.symbol in available_elements
+            ]
+        )
+
+        element_and_ch_treatment_options = {}
+        for element in elements_to_select:
+            if element in xch_elements:
+                element_and_ch_treatment_options[element] = "xch_smear"
+            else:
+                element_and_ch_treatment_options[element] = "full"
+
+        self.element_and_ch_treatment_options = element_and_ch_treatment_options
```

Comment:
Hi @superstar54. Does this seem like a reasonable way to address your [previous comment](https://github.com/aiidalab/aiidalab-qe/pull/580/files/3f6acbceb603cc282e1333674e3b4b7ddb0f84a2#r1465091243)?

I suggest this because the logic to define the initial values for the `element_and_ch_treatment` panel has already been defined in `_update_element_select_panel()`. This works fine with my own testing and sets the right behaviour for `set_panel_value()`.

```suggestion
        self._update_element_select_panel()
```